.github/workflows/docker-multistage-build-kops.yml

```diff
@@ -0,0 +1,172 @@
+---
+name: "Build multistage docker images (multi -flavours, -versions, -architectures -kops)"
+
+on:
+  workflow_call:
+    ###
+    ### Variables
+    ###
+    inputs:
+      matrix:
+        description: 'The build matrix'
+        required: true
+        type: string
+      stage:
+        description: 'The stage to build (Examples: base, mods, prod or work).'
+        required: true
+        type: string
+      stage_prev:
+        description: 'The previous stage (used for downloading previous artifact).'
+        required: true
+        type: string
+      artifact_prefix:
+        description: 'Unique artifact name prefix (to avoid overriding existing artifcats during parallel runs).'
+        required: true
+        type: string
+      has_refs:
+        description: 'The ref build matrix as JSON string (list of git refs to build/deploy).'
+        required: true
+        type: boolean
+      run_tests:
+        description: 'Dertermines whether we run integration tests or not.'
+        required: true
+        type: boolean
+      upload_artifact:
+        description: 'Dertermines whether we upload the artifact not.'
+        required: true
+        type: boolean
+      pull_base_image:
+        description: 'Pull Docker base image before building?.'
+        required: false
+        default: false
+        type: boolean
+
+jobs:
+  # -----------------------------------------------------------------------------------------------
+  # JOB: BUILD
+  # -----------------------------------------------------------------------------------------------
+  build:
+    name: ${{ matrix.name }}-${{ matrix.version }}-${{ inputs.stage }}${{ matrix.kops }} (${{ matrix.arch }}) ${{ matrix.refs }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include: ${{ fromJson(inputs.matrix) }}
+    steps:
+
+      # ------------------------------------------------------------
+      # Setup repository
+      # ------------------------------------------------------------
+      - name: "[SETUP] Checkout repository (current)"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+        if: ${{ !inputs.has_refs }}
+
+      - name: "[SETUP] Checkout repository (ref: ${{ matrix.refs }})"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ matrix.refs }}
+        if: ${{ inputs.has_refs }}
+
+      - name: "[SETUP] Install QEMU environment"
+        uses: docker/setup-qemu-action@v2
+        id: qemu
+        with:
+          image: tonistiigi/binfmt:latest
+          platforms: all
+
+      - name: "[SETUP] Install Docker Buildx"
+        id: buildx
+        uses: docker/setup-buildx-action@v2
```

Comment:
Build uses `docker build` and not buildx so likely this one isn't needed.

qemu+binfmt misc above should be enough.